core/src/main/scala/org/apache/spark/deploy/ExternalShuffleServiceSource.scala

```diff
@@ -19,19 +19,19 @@ package org.apache.spark.deploy
 
 import javax.annotation.concurrent.ThreadSafe
 
-import com.codahale.metrics.MetricRegistry
+import com.codahale.metrics.{MetricRegistry, MetricSet}
 
 import org.apache.spark.metrics.source.Source
-import org.apache.spark.network.shuffle.ExternalShuffleBlockHandler
 
 /**
  * Provides metrics source for external shuffle service
  */
 @ThreadSafe
-private class ExternalShuffleServiceSource
-(blockHandler: ExternalShuffleBlockHandler) extends Source {
+private class ExternalShuffleServiceSource extends Source {
```

Comment:
just for my own understanding, not directly related to this change --

I hadn't realized that the ExternalShuffleBlockHandler had its own ShuffleMetrics already.  Some of those metrics really seem like they should be part of regular shuffle server, in the executor.  Eg., [openBlockRequestLatencyMillis](https://github.com/apache/spark/blob/master/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java#L89).  Do you know why its separate?

Reply:
I'm not sure, maybe it should a part of regular shuffle server.